scripts/exregional_run_analysis.sh

```diff
@@ -0,0 +1,594 @@
+#
+START_DATE=`echo "${CDATE}" | sed 's/\([[:digit:]]\{2\}\)$/ \1/'`
+
+YYYYMMDDHH=`date +%Y%m%d%H -d "${START_DATE}"`
+JJJ=`date +%j -d "${START_DATE}"`
+
+YYYYMMDDHHmInterv=`date +%Y%m%d%H -d "${START_DATE} ${DA_CYCLE_INTERV} hours ago"`
+JJJm6=`date +%j -d "${START_DATE} ${DA_CYCLE_INTERV} hours ago"`
+
+YYYY=${YYYYMMDDHH:0:4}
+MM=${YYYYMMDDHH:4:2}
+DD=${YYYYMMDDHH:6:2}
+HH=${YYYYMMDDHH:8:2}
+YYYYMMDD=${YYYYMMDDHH:0:8}
+#
+#-----------------------------------------------------------------------
+#
+# go to working directory.
+# define fix and background path
+#
+#-----------------------------------------------------------------------
+
+cd_vrfy ${ANALWORKDIR}
+
+fixdir=$FIXgsi
+if [ ${BKTYPE} -eq 1 ]; then  # Use background from INPUT
+  bkpath=${CYCLE_DIR}/INPUT
+else
+  bkpath=${CYCLE_ROOT}/${YYYYMMDDHHmInterv}/RESTART
+fi
+
+print_info_msg "$VERBOSE" "fixdir is $fixdir"
+print_info_msg "$VERBOSE" "bkpath is $bkpath"
+
+
+#-----------------------------------------------------------------------
+#
+# Make a list of the latest GFS EnKF ensemble
+#
+#-----------------------------------------------------------------------
+
+stampcycle=`date -d "${START_DATE}" +%s`
+minHourDiff=100
+loops="009"
+for loop in $loops; do
+  for timelist in `ls ${ENKF_FCST}/*.gdas.t*z.atmf${loop}s.mem080.nemsio`; do
+    availtimeyy=`basename ${timelist} | cut -c 1-2`
+    availtimeyyyy=20${availtimeyy}
+    availtimejjj=`basename ${timelist} | cut -c 3-5`
+    availtimemm=`date -d "${availtimeyyyy}0101 +$(( 10#${availtimejjj} - 1 )) days" +%m`
+    availtimedd=`date -d "${availtimeyyyy}0101 +$(( 10#${availtimejjj} - 1 )) days" +%d`
+    availtimehh=`basename ${timelist} | cut -c 6-7`
+    availtime=${availtimeyyyy}${availtimemm}${availtimedd}${availtimehh}
+    AVAIL_TIME=`echo "${availtime}" | sed 's/\([[:digit:]]\{2\}\)$/ \1/'`
+    AVAIL_TIME=`date -d "${AVAIL_TIME}"`
+
+    stamp_avail=`date -d "${AVAIL_TIME} ${loop} hours" +%s`
+
+    hourDiff=`echo "($stampcycle - $stamp_avail) / (60 * 60 )" | bc`;
+    if [[ ${stampcycle} -lt ${stamp_avail} ]]; then
+       hourDiff=`echo "($stamp_avail - $stampcycle) / (60 * 60 )" | bc`;
+    fi
+
+    if [[ ${hourDiff} -lt ${minHourDiff} ]]; then
+       minHourDiff=${hourDiff}
+       enkfcstname=${availtimeyy}${availtimejjj}${availtimehh}00.gdas.t${availtimehh}z.atmf${loop}s
+    fi
+  done
+done
+
+size=`du --apparent-size --block-size=1 --dereference ${ENKF_FCST}/${enkfcstname}.mem001.nemsio | grep -o '..........' | sed -n '1 p'`
+ls ${ENKF_FCST}/${enkfcstname}.mem001.nemsio > filelist03
+
+n=2
+while [[ $n -le 80 ]] ; do
+  
+  if [ $n -lt 10  ]; then
+    nn=00$n
+  else
+    nn=0$n
+  fi
+
+  size2=`du --apparent-size --block-size=1 --dereference ${ENKF_FCST}/${enkfcstname}.mem${nn}.nemsio | grep -o '..........' | sed -n '1 p'`
+
+  if [[ $size2 -lt $size ]]; then
+    print_info_msg "$VERBOSE"  "Bad GDAS member number ${nn}."
+  else
+    ls ${ENKF_FCST}/${enkfcstname}.mem${nn}.nemsio >> filelist03
+  fi
```

Comment:
The check on file size isn't robust to a bad member001.

Reply:
Copied this from RAP/HRRR and we can improve it later when we have better idea.